django/db/models/fields/related.py

```diff
@@ -1885,6 +1899,30 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required kwarg for %s in Django 2.0. "
+                "Set it to models.CASCADE if you want to maintain the current default behavior. "
+                "See https://docs.djangoproject.com/en/%s/ref/models/fields/"
+                "#django.db.models.ForeignKey.on_delete" % (
+                    self.__class__.__name__,
+                    get_docs_version(),
+                ),
+                RemovedInDjango20Warning, 2)
+            on_delete = CASCADE
+
+        if not callable(on_delete) and callable(to_field):
```

Comment:
Thanks for the clarity, Tim :-)

In that case, I think we may as well still go ahead and provide the deprecation shim and warning that we are able to provide (for `ForeignKey(SomeModel, 'to_field')`, which is likely much more common) by shortening this line to just `if not callable(on_delete)`. And we'll just have to rely on a backwards-incompatibility note in the release notes (that you should no longer pass `to_field` positionally, and you can pass `on_delete` as the positional second arg) to help anyone who is doing `ForeignKey(SomeModel, 'to_field', on_delete=models.WHATEVER)`.